pkg/series/storeapi/storeapi.go

```diff
@@ -91,6 +91,10 @@ func (i *iterator) Next() bool {
 	}
 
 	i.currentSeries = seriesResp.GetSeries()
```

Comment:
Hey, nil is actually ok because we have different responses: warnings hints as well..

We should gather warnings at least and log them and otherwise ignore non series frames (:

Reply:
umm what do you suggest how do we prevent Seg faults then?

Reply:
As in 99% cases you can always check how we deal with this on Thanos (: 

https://github.com/thanos-io/thanos/blob/49dad93ed67210374939aedd7a8f849a60375906/pkg/query/querier.go#L153